pkg/controller/volume/attach_detach_controller.go

```diff
@@ -106,30 +240,364 @@ func (adc *attachDetachController) Run(stopCh <-chan struct{}) {
 		go adc.internalNodeInformer.Run(stopCh)
 	}
 
+	if adc.selfCreatedPVCInformer {
+		go adc.internalPVCInformer.Run(stopCh)
+	}
+
+	if adc.selfCreatedPVInformer {
+		go adc.internalPVInformer.Run(stopCh)
+	}
+
+	go adc.reconciler.Run(stopCh)
+
 	<-stopCh
 	glog.Infof("Shutting down Attach Detach Controller")
 }
 
 func (adc *attachDetachController) podAdd(obj interface{}) {
-	// No op for now
+	pod, ok := obj.(*api.Pod)
+	if pod == nil || !ok {
+		return
+	}
+
+	addVolumes := pod.Status.Phase == api.PodPending ||
```

Comment:
Spoke offline: recommendation is to use `NodeName` instead. If it exists, the pod is scheduled and volumes should be attached. Will make this change.